tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/stomp/StompAuditLoggingTest.java

```diff
@@ -0,0 +1,107 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.activemq.artemis.tests.integration.stomp;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import org.apache.activemq.artemis.core.protocol.stomp.Stomp;
+import org.apache.activemq.artemis.core.security.Role;
+import org.apache.activemq.artemis.core.server.ActiveMQServer;
+import org.apache.activemq.artemis.logs.AssertionLoggerHandler;
+import org.apache.activemq.artemis.logs.AssertionLoggerHandler.LogLevel;
+import org.apache.activemq.artemis.logs.AuditLogger;
+import org.apache.activemq.artemis.spi.core.security.ActiveMQJAASSecurityManager;
+import org.apache.activemq.artemis.tests.integration.stomp.util.ClientStompFrame;
+import org.apache.activemq.artemis.tests.integration.stomp.util.StompClientConnection;
+import org.apache.activemq.artemis.tests.integration.stomp.util.StompClientConnectionFactory;
+import org.apache.activemq.artemis.tests.util.RandomUtil;
+import org.apache.activemq.artemis.tests.util.Wait;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class StompAuditLoggingTest extends StompTestBase {
+
+   private static final String BASE_AUDIT_LOGGER_NAME = AuditLogger.BASE_LOGGER.getLogger().getName();
+   private static LogLevel previousLevel = null;
+   private static AssertionLoggerHandler loggerHandler;
+   protected StompClientConnection conn;
+   private final String user = "nopriv";
+   private final String pass = user;
+   private final String role = "nopriv";
+
+   @Override
+   public boolean isSecurityEnabled() {
+      return true;
+   }
+
+   @Override
+   protected ActiveMQServer createServer() throws Exception {
+      server = super.createServer();
+
+      ActiveMQJAASSecurityManager securityManager = (ActiveMQJAASSecurityManager) server.getSecurityManager();
+
+      securityManager.getConfiguration().addUser(user, pass);
+      securityManager.getConfiguration().addRole(user, role);
+      server.getConfiguration().getSecurityRoles().put("#", new HashSet<>(Set.of(new Role(role, false, false, false, false, false, false, false, false, false, false))));
+
+      return server;
+   }
+
+   @Override
+   @Before
+   public void setUp() throws Exception {
+      super.setUp();
+      conn = StompClientConnectionFactory.createClientConnection(uri);
+   }
+
+   @BeforeClass
+   public static void prepareLogger() {
+      previousLevel = AssertionLoggerHandler.setLevel(BASE_AUDIT_LOGGER_NAME, LogLevel.INFO);
+      loggerHandler = new AssertionLoggerHandler();
+   }
+
+   @AfterClass
+   public static void clearLogger() throws Exception {
+      try {
+         loggerHandler.close();
+      } finally {
+         AssertionLoggerHandler.setLevel(BASE_AUDIT_LOGGER_NAME, previousLevel);
+      }
+   }
+
+   @Test
+   public void testConcurrentLogging() throws Exception {
```

Comment:
Not really clear from the test which bit is concurrent / with what, i.e what is the interesting bit of what the test is actually doing? Perhaps add a comment?

Reply:
My bad. The test name was copied from elsewhere. I'll rename it to actually reflect what it's doing.